addon/components/au-file-upload.gts

```diff
@@ -55,36 +89,42 @@ export default class AuFileUpload extends Component {
     return this.uploadErrorData.length > 0;
   }
 
-  @task
-  *upload(file) {
+  @action
+  upload(file: UploadFile): void | undefined {
+    this.uploadTask.perform(file);
+  }
+
+  uploadTask = task(async (file: UploadFile) => {
     this.resetErrors();
-    let uploadedFile = yield this.uploadFileTask.perform(file);
+    let uploadedFile: number | null = await this.uploadFileTask.perform(file);
```

Comment:
I _think_ we can type the return type of the  task instead. Not sure why I did this.